Add note about Dependabot promoting some of our dependencies (#37)
alphagov/design-system-team-docs: source/how-we-work/version-control/pull-requests.html.md

If you want to check the changes being made to the dependency, use a tool like [Package Diff](https://diff.intrinsic.com/) which shows the difference between the published packages.

> A quick look at the `package-lock.json` file may reveal that Dependabot promoted some of our `devDependencies` to `dependencies`. If it occurs, this can usually be fixed by checking out the branch, running `npm install` and pushing a new commit with the resulting `package-lock.json`.

Reviewer: This either only affects `optionalDependencies` or it's a bug limited to `optionalDependencies` that happen to already be installed (by one of our workspaces or an existing `node_modules` dependency)

```suggestion
A quick look at the `package-lock.json` file may reveal that Dependabot has incorrectly duplicated some `optionalDependencies` to `dependencies`. If it occurs, this can usually be fixed by checking out the branch, running `npm install` and pushing a new commit with the resulting `package-lock.json`.
```
